healthcheck.go

```diff
@@ -0,0 +1,117 @@
+package hc
+
+import (
+	aah "aahframe.work"
+	"sync"
+	"time"
+)
+
+// HealthCheckReporter contains the dependencies to check
+type HealthCheckReporter struct {
+	dependencies    map[string]Dependency
+	globalHealth    bool
+	globalHealthMsg map[string]string
+	mu              sync.RWMutex
+}
+
+// HealthChecker is the interface for a Dependency that can be health-checked
+type HealthChecker interface {
+	// Check will return nil if dependency is reachable/healthy
+	// You should implement this func with a sensible timeout (< 3 or 5 sec)
+	Check() error
+}
+
+// Dependency implements the HealthChecker interfance and also contains the check config
+type Dependency struct {
+	// implements HealthChecker interface
+	HealthChecker
+	// softDep - if true it will allow errors so won't report unhealthy
+	softDep bool
+	// TODO: interval unused for now - got a global interval for all checks
```

Comment:
not sure if each dependency should have its own interval rather than the Reporter checking all of them globally with the same interval?

Reply:
We can start with the global interval for initial release later we could provide for each based on user request.